# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/tabs/src/sync/engine.rs
# Commit: 57b2cda7e0db
# Full Hash: 57b2cda7e0db8b07fcc1e09113ce26add178c68a
# Author: Sammy Khamis <skhamis@mozilla.com>
# Date: 2023-03-01 03:56:13
# Regressor Bug: 1818349
# File Overlap Count: 23
# Overlapping Files: third_party/rust/sync15/src/client/coll_update.rs, third_party/rust/sync15/src/client/mod.rs, third_party/rust/sync15/.cargo-checksum.json, third_party/rust/sync15/src/client/sync.rs, Cargo.toml
# Description:
#   Bug 1818349 Part 1: Vendor new version of application-services r=markh
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D171181
# ==============================================================================

diff -r eaf9f12a1481 -r 57b2cda7e0db third_party/rust/tabs/src/sync/engine.rs
--- a/third_party/rust/tabs/src/sync/engine.rs	Tue Feb 28 04:22:35 2023 +0000
+++ b/third_party/rust/tabs/src/sync/engine.rs	Tue Feb 28 05:14:29 2023 +0000
@@ -311,11 +311,7 @@
             .unwrap()
             .apply_incoming(inbound.changes, telem)?;
 
-        Ok(OutgoingChangeset::new_with_changes(
-            "tabs",
-            inbound.timestamp,
-            outgoing_records,
-        ))
+        Ok(OutgoingChangeset::new("tabs".into(), outgoing_records))
     }
 
     fn sync_finished(
@@ -343,7 +339,9 @@
         Ok(if since == server_timestamp {
             vec![]
         } else {
-            vec![CollectionRequest::new("tabs").full().newer_than(since)]
+            vec![CollectionRequest::new("tabs".into())
+                .full()
+                .newer_than(since)]
         })
     }
 